.circleci/config.yml

```diff
@@ -1,5 +1,171 @@
-version: 2
+version: 2.1
+
+executors:
+  go:
+    working_directory: ~/go/src/github.com/hashicorp/nomad
+    docker:
+      - image: dantoml/nomad-build-image:e8d69b1e8
+  go-machine:
+    working_directory: ~/go/src/github.com/hashicorp/nomad
+    machine:
+      image: circleci/classic:201808-01
+  docker-builder:
+    working_directory: ~/go/src/github.com/hashicorp/nomad
+    machine: true # TODO: Find latest docker image id
+
 jobs:
+  build-deps-image:
+    executor: docker-builder
+    steps:
+      - checkout
+      - run: docker pull dantoml/nomad-build-image:latest || true
+      - run: docker build -t dantoml/nomad-build-image:$CIRCLE_SHA1 ./.circleci/build-images/go
+      - run: docker push dantoml/nomad-build-image:$CIRCLE_SHA1
+
+  lint:
+    executor: go
+    environment:
+      GOPATH: /home/circleci/go
+    steps:
+      - checkout
+      - run: make check
+
+  test-container:
+    executor: go
+    parameters:
+      test_packages:
+        type: string
+        default: ""
+      exclude_packages:
+        type: string
+        default: ""
+    environment:
+      GOTEST_PKGS: "<< parameters.test_packages >>"
+      GOTEST_PKGS_EXCLUDE: "<< parameters.exclude_packages >>"
+      GOPATH: /home/circleci/go
+      GOTESTSUM_JUNITFILE: /tmp/results.xml
+      GOMAXPROCS: 1
+      NOMAD_SLOW_TEST: 1
+    steps:
+      - checkout
+      - run: |
+          if [ -z $GOTEST_PKGS_EXCLUDE ];
+          then
+            unset GOTEST_PKGS_EXCLUDE
+          else
+            unset GOTEST_PKGS
+          fi
+          make test-nomad
```

Comment:
In Travis, we also run `generate-structs` to exercise tests with code generated structs encoders, as done in final releases.